Descent3/menu.cpp

```diff
@@ -1398,6 +1455,76 @@ int DisplayLevelWarpDlg(int max_level) {
   hwnd.Destroy();
   return chosen_level;
 }
+
+// DisplayLevelSelectDlg
+// displays a list of levels associated with the selected mission
+int DisplayLevelSelectDlg(int max_level) {
+  newuiTiledWindow lvlsel;
+  newuiSheet *select_sheet;
+  newuiListBox *level_lb;
+  level_info lvinfo;
+  char level_name[100];
+  int chosen_level = 1, res;
+  char buffer[128];
+  int highest_allowed;
+
+  // Create the menu
+  lvlsel.Create(msnname, 0, 0, 448, 384);
+  select_sheet = lvlsel.GetSheet();
+  select_sheet->NewGroup(NULL, 10, 0);
+  level_lb = select_sheet->AddListBox(352, 256, UID_LVLB, UILB_NOSORT);
+  select_sheet->NewGroup(NULL, 160, 280, NEWUI_ALIGN_HORIZ);
+  select_sheet->AddButton(TXT_OK, UID_OK);
+  select_sheet->AddButton(TXT_CANCEL, UID_CANCEL);
+
+  // Loop through the levels
+  for (int level = 1; level <= max_level; ++level) {
+    // Check if msnname matches core mission names, if so use the arrays for level names
+    // Otherwise, try to get the level name from the level info
+    if (strcmp(msnname, "Descent 3: Retribution") == 0) {
+      if (level >= 1 && level <= sizeof(RetributionNames) / sizeof(RetributionNames[0])) {
+        if (level > 15) { // If they're the secret levels we use a different format
+          snprintf(level_name, sizeof(level_name), "Secret Lv - %s", RetributionNames[level - 1]);
+        } else {
+          snprintf(level_name, sizeof(level_name), "Lv %02d - %s", level, RetributionNames[level - 1]);
```

Comment:
Better to avoid `snprintf()` usage here, just place "Lv <num> - <name>" or "Secret Lv - <name>" into mission names lists, and this code will be simpler.

Reply:
I'm not going to attempt that, since I don't have experience with the newer standard, and there are other loops in this file that are similar. If you want to simplify the entire menu.cpp file this way, by all means feel free to merge this pr into a new one that actually cleans up the whole file.

Reply:
I looked into this some more, and while I can add Lv## to the array easily to simplify this, the problem comes when the array is not used and we then need snprintf for the level number. I think this is easier. If you refactor the whole menu.cpp file in the future, this could certainly qualify for a pass.